### PR TITLE
Initial dockerpty setup

### PR DESCRIPTION
**Pull request by mjtravers**
This is not an actual pull request yet. Just sharing code.

The tty terminal replays all the commands run from a previous session when the container is restarted. Still figuring that one out.

To login to a docker container, first create it and then login (like so):

niceman create -r my-container -t docker-container
niceman login -r my-container

### CODE REVIEW

**Comment by satra** [comment]
@mjtravers - that would be a very nice functionality

**Comment by mjtravers** [comment]
Yarik,

I'm done doing all I think I am doing for this pull request and it is passing the tests.

Feel free to critique and/or reject and/or merge.

**Comment by yarikoptic** [comment]
BTW, read the readme for dockerpty -- `C-p C-q` pretty much detaches from the session so if you hit enter, it would crash and exit but then would leave docker running.  not sure yet on how/either we should (ab)use that "feature" ;)

File: niceman/interface/login.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import re
+
+from .base import Interface, get_resource_info
+import niceman.interface.base # Needed for test patching
+from ..support.param import Parameter
+from ..support.constraints import EnsureStr
+from ..resource import Resource
+
+from logging import getLogger
+lgr = getLogger('niceman.api.delete')
```
Comment (yarikoptic) [inline niceman/interface/login.py]: ok -- I guess we will fix all those logging targets later anyways

**Pull request merged**